pkg/datapath/linux/node_linux_test.go

```diff
@@ -94,8 +94,14 @@ var (
 
 	mcastNumIPv4 = []string{"net", "ipv4", "neigh", "default", "mcast_solicit"}
 	mcastNumIPv6 = []string{"net", "ipv6", "neigh", "default", "mcast_solicit"}
+	spi          = uint8(0)
```

Comment:
Is this really necessary? This sort of global variable in tests can easily create side effects that then end up introducing flakes.

Reply:
AFAIU, in these tests we just need an IPSEC key to be present, and we never remove it between these tests.
Possible solutions:

1. Let the tests call the `LoadIPSecKey` with an incremental SPI each time.
2. Let the tests call the `LoadIPSecKey` only once.

Either way, instead of global var, we could wrap the incremental SPI or a boolean variable. Would this be a preferred way? 

```go
var loadOnceIPSecKey func(t *testing.T, log *slog.Logger) = func() func(t *testing.T, log *slog.Logger) {
	var loaded bool
	inner := func(t *testing.T, log *slog.Logger) {
		if loaded {
			return
		}
		keys := bytes.NewReader([]byte("6 rfc4106(gcm(aes)) 44434241343332312423222114131211f4f3f2f1 128\n"))
		_, _, err := ipsec.LoadIPSecKeys(log, keys)
		require.NoError(t, err)
		loaded = true
	}
	return inner
}()
```

Reply:
Could we clean up the key after each test? That way we avoid state leaking from one test to the next. Then it doesn't matter what each test does because they're all independent.

Reply:
I'm scared we don't have a method yet to clean up the key held in `ipSecKeysGlobal`, unless providing a new key with different SPI. Am I missing something? In case, would it be worth adding? 
Otherwise, for the just sake of this test a `sync.Once` could be enough, but I do recognize that what you're suggesting would be a way more elegant solution.

Reply:
I'm just suggesting to do the cleanup from `setupIPSecSuitePrivileged` after each test instead of once at the end. Am I missing something?

Reply:
I think I got what you meant, hopefully :smiley:

Reply:
IIUC, shouldn't the new API you added, `UnsetTestIPSecKey()`, be called from `setupIPSecSuitePrivileged` as well? Now it's only called in the node test suite.

Reply:
> IIUC, shouldn't the new API you added, UnsetTestIPSecKey(), be called from setupIPSecSuitePrivileged as well? Now it's only called in the node test suite.

Since `setupIPSecSuitePrivileged` is within the same package, in theory it wouldn't be necessary to invoke the public API (so far it directly reinitialize the `ipSecKeysGlobal`). What would you suggest?

Reply:
I don't have a strong opinion on this but since we have a public API that is needed outside the package I don't see why we shouldn't use it inside the package itself to make the cleanup of the key consistent in and outside of the ipsec package?

Reply:
I see your point, good to have coherent behavior especially while tearing down test suites. Modified the latest commit, let me know :smiley:

Reply:
LGTM. I'll remove the discussion label so we can get the PR merged. If Paul still sees something that is missing we can address it in a follow up I presume.